docs/CHANGELOG.md

```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Added `hexencode` and `base64encode` tasks (pipeline).
 - `forwardingAllowed` per job attribute to allow forwarding txs submitted by the job.
+- `NODE_SELECTION_MODE` (`EVM.NodePool.SelectionMode`) controls node picking strategy. Supported values: `HighestHead` and `RoundRobin` (default).
```

Comment:
This is a much-requested feature, maybe worth extending this description a bit to explain what it does.

Reply:
Did we consider making `HighestHead` the default? I understand if we want to introduce it carefully, but IMHO it is the much safer/stabler choice

Reply:
I believe it will be safer to release with RoundRobin=default, then after we test HighestHead in action in prod, we can make it default in the future releases.

Reply:
What sort of testing, other than the usual release testing?
IMHO this is not only an improvement, but arguably a bug fix for ongoing issues, and it would be unfortunate to slow this down across multiple release cycles

Reply:
The source of concern here is the ask to keep `RoundRobin` option, which means we are not 100% sure the new mode will always work best. For testing, I meant for example VRF can try enable this and see if this fixes the problem on Goerli or does not. 
I think for the next release we should implement `eth_syncing`, and, as long as `HighestHead` is working fine, remove `RoundRobin` and the selection option (env var).

Reply:
I think eth_syncing is an orthogonal concern. If a node reports that kind of syncing, it is simply ineligible to be used at all, the same as if it returned the wrong chain ID.

Reply:
> which means we are not 100% sure the new mode will always work best. 

I disagree. IMHO we know 100% that this will be an iterative improvement - the only thing I see to test is whether it works or has bugs. Furthermore, we can continue to iterate with even better versions. For example, it would be even more stable to stick to one node until it goes bad, rather than flapping between nodes that briefly get ahead of one another. This would also restore the assumption that our systems were already depending on: that the next RPC call will be made to the same endpoint as the previous, and which is subscribed to.

Reply:
Agreed. https://github.com/smartcontractkit/chainlink/pull/7268

Reply:
Agreed. I made `HighestHead` to be default now.
There is one edge case (bug) I found: `HighestHead` will work right only if "no new heads threshold" is enabled. I made a note in CHANGELOG and added an error message logged if it is not enabled in this combination.
However, I would like to hear your opinion - maybe we should handle this case differently?

Reply:
I suspect we could change things to make it compatible, but I'm not sure that is worth delaying much longer.